src/ios/CodePush.m

```diff
@@ -224,45 +224,45 @@ - (void)dealloc {
     [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
-- (void)handleAppStart {
+- (void)clearDeploymentsIfBinaryUpdated {
     // check if we have a deployed package
     CodePushPackageMetadata* deployedPackageMetadata = [CodePushPackageManager getCurrentPackageMetadata];
     if (deployedPackageMetadata) {
         NSString* deployedPackageNativeBuildTime = deployedPackageMetadata.nativeBuildTime;
         NSString* applicationBuildTime = [Utilities getApplicationTimestamp];
 
         if (deployedPackageNativeBuildTime != nil && applicationBuildTime != nil) {
-            if ([deployedPackageNativeBuildTime isEqualToString: applicationBuildTime] ) {
-                // same version, safe to launch from local storage
-                if (deployedPackageMetadata.localPath) {
-                    [self redirectStartPageToURL: deployedPackageMetadata.localPath];
-                }
-            }
-            else {
-                // installed native version is different from package version => clean up deployed package and do not modify start page
+            if (![deployedPackageNativeBuildTime isEqualToString: applicationBuildTime]) {
```

Comment:
(i don't know if Objective-C's `isEqualToString` has the same behavior as Java's `.equals` but i would look into it)